dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -286,6 +286,15 @@ AppNotificationAssets Microsoft::Windows::AppNotifications::Helpers::GetAssets()
     return assets;
 }
 
+AppNotificationAssets Microsoft::Windows::AppNotifications::Helpers::ValidateAssets(std::wstring const& displayName, std::filesystem::path iconFilePath)
+{
+    winrt::check_bool(std::filesystem::exists(iconFilePath));
+
+    THROW_HR_IF_MSG(E_INVALIDARG, !IsIconFileExtensionSupported(iconFilePath), "Icon format not supported");
+
+    return AppNotificationAssets{ std::move(displayName), iconFilePath.wstring() };
```

Comment:
Is the ctor using an R-Value ref? If not, why is std::move needed if it is simply a pass by ref?

Reply:
Ok looks like you don't need r-value semantics to use std::move. So this should technically work.
https://stackoverflow.com/questions/59553364/c-using-move-semantics-to-pass-argument-to-another-function

Reply:
This is the key:
Note on move semantics

If you use move semantics, the arguments of your functions don't necessarily need the double &&. Generally, it is better to omit them (except for perfect forwarding and non-copyable objects like stream objects). The && requires that the passed arguments be rvalues. However, omitting the && does not mean you cannot pass the arguments as rvalues. Let's see this with an example: